x-pack/plugins/uptime/public/components/synthetics/check_steps/steps_list.tsx

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiBasicTable, EuiPanel, EuiTitle } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React, { useCallback, useState, useEffect, MouseEvent } from 'react';
+import styled from 'styled-components';
+import { useDispatch } from 'react-redux';
+import { Ping } from '../../../../common/runtime_types';
+import { pruneJourneyState } from '../../../state/actions/journey';
+import { clearPings } from '../../../state/actions';
+import { STATUS_LABEL } from '../../monitor/ping_list/translations';
+import { STEP_NAME_LABEL } from '../translations';
+import { ExpandRowColumn, toggleExpand } from './expand_row_col';
+import { StatusBadge } from '../status_badge';
+import { StepDetailLink } from '../../common/step_detail_link';
+import { VIEW_PERFORMANCE } from '../../monitor/synthetics/translations';
+import { StepImage } from './step_image';
+
+export const SpanWithMargin = styled.span`
+  margin-right: 16px;
+`;
+interface Props {
+  data: Ping[];
+  error?: Error;
+  loading: boolean;
+}
+
+function isStepEnd(step: Ping) {
+  return step.synthetics?.type === 'step/end';
+}
+interface StepStatusCount {
+  failed: number;
+  skipped: number;
+  succeeded: number;
+}
+
+function statusMessage(count: StepStatusCount) {
+  const total = count.succeeded + count.failed + count.skipped;
+  if (count.failed + count.skipped === total) {
+    return i18n.translate('xpack.uptime.synthetics.journey.allFailedMessage', {
+      defaultMessage: '{total} Steps - all failed or skipped',
+      values: { total },
+    });
+  } else if (count.succeeded === total) {
+    return i18n.translate('xpack.uptime.synthetics.journey.allSucceededMessage', {
+      defaultMessage: '{total} Steps - all succeeded',
+      values: { total },
+    });
+  }
+  return i18n.translate('xpack.uptime.synthetics.journey.partialSuccessMessage', {
+    defaultMessage: '{total} Steps - {succeeded} succeeded',
+    values: { succeeded: count.succeeded, total },
+  });
+}
+
+function reduceStepStatus(prev: StepStatusCount, cur: Ping): StepStatusCount {
+  if (cur.synthetics?.payload?.status === 'succeeded') {
+    prev.succeeded += 1;
+    return prev;
+  } else if (cur.synthetics?.payload?.status === 'skipped') {
+    prev.skipped += 1;
+    return prev;
+  }
+  prev.failed += 1;
+  return prev;
+}
```

Comment:
@paulb-elastic i preserved this heading logic from ping list 

it's written by @justinkambic , i think it works pretty great

Reply:
One thing we could do to simplify a bit is nest `prev.failed += 1` in an `else` block and remove the first two `return`s, so we'd only have one exit point.